### DISCUSSION TITLE
fcitx5 在weston里启动失败

### DISCUSSION THREAD

**Discussion by guyuming76** [fcitx/fcitx5#1018] [Q&A] [answered]
我在x11下使用x11 backend启动weston, 然后在weston里启动fcitx5不成功,下面链接是通过script 命令获取的输出:

https://pastebin.com/1HRDakqU

[fcitx5_out.txt](https://github.com/fcitx/fcitx5/files/15007514/fcitx5_out.txt)

这个可能是啥原因,如何定位到代码?

**Comment by wengxt** [comment]
你怎么在 weston 里启动 fcitx5 的

**Reply by guyuming76** [reply]
我打开weston-terminal后,运行 fcitx5 --disable=dbus 命令或者 fcitx5 --disable=dbus --verbose "*=5"

**Reply by wengxt** [reply, marked as answer]
这样总而言之是用不了 weston 的输入法支持的。

应该是因为 fcitx5 尝试初始化 wayland 输入法但是并不允许。 

请参考 https://fcitx-im.org/wiki/Using_Fcitx_5_on_Wayland#Weston 进行配置，或者禁用 waylandim 。

另外，如果你想要使用已经启动的 fcitx5，推荐 path=/usr/libexec/fcitx5-wayland-launcher 没必要启动一个禁用 dbus 的，而且同时运行两个 fcitx 会同时写入数据，并不好。

https://www.csslayer.info/wordpress/fcitx-dev/fcitx-5-and-multiple-wayland-compositor/

**Reply by guyuming76** [reply]
我刚查了下weston源码, 在 b57f472c8  和 677a5f5ac 这两个commit里面 完成了启动compositor后 bind_input_panel, 即zwp_input_panel_v1. 也许fcitx5启动时又要bind,所以报 interface object already bound 了.

这个东西一直是这样? 我平时主要用基于wlroots的dwl,很少用weston,不记得之前有没有遇到到这个问题

**Reply by wengxt** [reply]
我觉得你看见的错误信息不是实际的原因。weston 只有在配置文件里面设置的输入法才能使用这个对象

你指的代码都十年前的 ´_>` ……怎么可能有关系

**Reply by wengxt** [reply]
这是一种为了安全的机制，否则像 wlroots 那样什么进程都能成为输入法，就可以轻易监听所有键盘输入了
